core/src/main/java/org/elasticsearch/rest/action/admin/cluster/snapshots/get/RestGetSnapshotsAction.java

```diff
@@ -47,7 +47,12 @@ public RestGetSnapshotsAction(Settings settings, RestController controller, Clie
     public void handleRequest(final RestRequest request, final RestChannel channel, final Client client) {
         String repository = request.param("repository");
         String[] snapshots = request.paramAsStringArray("snapshot", Strings.EMPTY_ARRAY);
-        GetSnapshotsRequest getSnapshotsRequest = getSnapshotsRequest(repository).snapshots(snapshots);
+        boolean ignoreUnavailable = request.paramAsBoolean("ignore_unavailable", false);
```

Comment:
Might be better to use the default of the request (in this case this coincides, but explicit is better in case of refactoring):

getSnapshotsRequest.ignoreUnavailable(request.paramAsBoolean("ignore_unavailable", getSnapshotsRequest.ignoreUnavailable());